src/idm/authentication.js

```diff
@@ -0,0 +1,57 @@
+import axios from 'axios';
+import qs from 'qs';
+import { errorHandler } from '../utils';
+
+const entity = (base, token) => {
+  base = `${base}`;
+
+  return ({
+    /**
+    * @summary Authenticate a client with client secret and client name.
+    * @name authenticateClient
+    * @public
+    * @function
+    * @memberof agile.idm.authentication
+    * @param {String} client - client name. This is the client name provided to the create Entity when you register an Oauth2 client in AGILE-IDM. For more info: https://github.com/Agile-IoT/agile-idm-oauth2-client-example
+    * @param {String} secret - client secret. This is the client name provided to the create Entity when you register an Oauth2 client in AGILE-IDM. For more info: https://github.com/Agile-IoT/agile-idm-oauth2-client-example
+    * @fulfil {Object} Authentication information including token_type and access_token
+    * @returns {Promise}
+    * @example
+    * agile.idm.authentication.authenticateClient("client_name","credentials").then(function(result) {
```

Comment:
Actually no. In this case, credentials is a string with a client secret. I updated it to make it more like a random token in the example. I hope this makes it clearer now